source/backend/dal/Repositories/PropertyRepository.cs

```diff
@@ -326,8 +326,9 @@ public void Delete(PimsProperty property)
                 .Include(p => p.PimsPropertyAcquisitionFiles)
                 .FirstOrDefault();
 
-            existingProperty.PimsPropertyResearchFiles.ForEach(pr => this.Context.Remove(pr));
-            existingProperty.PimsPropertyAcquisitionFiles.ForEach(pa => this.Context.Remove(pa));
+            existingProperty.PimsPropertyResearchFiles.ForEach(pr => this.Context.Remove(new PimsPropertyResearchFile() { Id = pr.Id }));
+            existingProperty.PimsPropertyAcquisitionFiles.ForEach(pa => this.Context.Remove(new PimsPropertyAcquisitionFile() { Id = pa.Id }));
+            existingProperty.PimsPropertyLeases.ForEach(l => this.Context.Remove(new PimsPropertyLease() { Id = l.Id }));
```

Comment:
It's a bit odd that the "existingProperty" EF query doesn't include the related data for leases yet we try to delete them. Have you tested that "PimsPropertyLeases" is populated with the above query?

Github doesn't let me highlight the relevant code so here is a screenshot:

![image](https://user-images.githubusercontent.com/24322224/208993814-090bc256-1692-438e-941e-a9e438f4ac11.png)

Reply:
In my tests it was populated, but I agree that for consistency it should be included, will add that.